napari/utils/translations.py

```diff
@@ -274,33 +277,34 @@ def translation(self) -> str:
         """
         Return the translated string with interpolated kwargs, if provided.
         """
-
-        if self._n is None and self._msgctxt is None:
-            translation = gettext.dgettext(
-                self._domain,
-                self._msgid,
-            )
-        elif self._n is None:
-            translation = gettext.dpgettext(
-                self._domain,
-                self._msgctxt,
-                self._msgid,
-            )
-        elif self._msgctxt is None:
-            translation = gettext.dngettext(
-                self._domain,
-                self._msgid,
-                self._msgid_plural,
-                self._n,
-            )
-        else:
-            translation = gettext.dnpgettext(
-                self._domain,
-                self._msgctxt,
-                self._msgid,
-                self._msgid_plural,
-                self._n,
-            )
+        if self._n is None:
+            if self._msgctxt is None:
+                translation = gettext.dgettext(
+                    self._domain,
+                    self._msgid,
+                )
+            else:
+                translation = gettext.dpgettext(
+                    self._domain,
+                    self._msgctxt,
+                    self._msgid,
+                )
+        elif self._msgid_plural is not None:
+            if self._msgctxt is None:
+                translation = gettext.dngettext(
+                    self._domain,
+                    self._msgid,
+                    self._msgid_plural,
+                    self._n,
+                )
+            else:
+                translation = gettext.dnpgettext(
+                    self._domain,
+                    self._msgctxt,
+                    self._msgid,
+                    self._msgid_plural,
+                    self._n,
+                )
```

Comment:
These control flows allow having `translation` undefined. Maybe last else should be dedent

Reply:
I've tidied up the logic here to add an `else:` statement at the end, and hopefully make it a bit more readable.